### DISCUSSION TITLE
Ignore one dockerfile variant

### DISCUSSION THREAD

**Discussion by donny-dont** [renovatebot/renovate#8365] [Request Help]
Hi I have a repository of Dockerfiles <https://github.com/WebKitForWindows/docker-webkit-dev> that are building containers for different Windows versions. I used `fileMatch` because the I only want Renovate to look at files in the `base` directory and ignore the others in the repository. There is also a `Dockerfile.aws` that I don't want Renovate to auto update because we verify that base image works in an AMI.

I had the following but I saw that the `Dockerfile.aws` had been updated by Renovatebot so I'm wondering what the proper config should look like.

```json
{
  "extends": [
    "config:base",
    ":automergeDigest"
  ],
  "dockerfile": {
    "fileMatch": [
      "base/Dockerfile\\.(1803|1809|1903|1909|2004|windows-1809|windows-1903|windows-1909|windows-2004)"
    ],
    "pinDigests": true
  },
  "labels": [
    "renovate"
  ]
```

Thanks in advance for any help!

**Comment by rarkins** [comment]
Can you search your logs for fileMatch to verify that the pattern you intended is being used?

**Reply by donny-dont** [reply]
It doesn't appear to be used. I'm looking in https://app.renovatebot.com/dashboard#github/WebKitForWindows/docker-webkit-dev/289935716 and I see multiple matches that I would expect to be ignored.

The docs aren't super clear if `docker` or `dockerfile` should be used.

Do you have any particular suggestion on how to best write this config?

**Reply by rarkins** [reply]
I just remembered that fileMatch is additive, so defining a new pattern does not delete the existing. You'd need to use includePaths or ignorePaths instead of fileMatch

**Reply by donny-dont** [reply]
So something like this

```json
{
  "extends": [
    "config:base",
    ":automergeDigest"
  ],
  "dockerfile": {
    "includePaths": ["base/"],
    "ignorePaths": ["base/Dockerfile.aws"],
    "pinDigests": true
  },
  "labels": [
    "renovate"
  ]
}
```

**Reply by rarkins** [reply]
Yes, but I'm not 100% confident until it's tested as I've not tried using both at the same time before.

**Reply by donny-dont** [reply]
Ok cool I'll go ahead and commit it and see what happens.

There's no cli for debugging a config right? Would be nice to have a `renovate-lint` that'd parse the config and see what it would pick up with the repo.

**Reply by donny-dont** [reply]
Tried that and got https://app.renovatebot.com/dashboard#github/WebKitForWindows/docker-webkit-dev/289952007 so it doesn't appear to like having `includePaths` and `ignorePaths`.

Just using `ignorePaths` on the Dockerfile.aws and the other directories I was able to get the following output from https://app.renovatebot.com/dashboard#github/WebKitForWindows/docker-webkit-dev/289952070 .

```
DEBUG: Matched 11 file(s) for manager dockerfile: base/Dockerfile.1803, base/Dockerfile.1809, base/Dockerfile.1903, base/Dockerfile.1909, base/Dockerfile.2004, base/Dockerfile.20H2, base/Dockerfile.windows-1809, base/Dockerfile.windows-1903, base/Dockerfile.windows-1909, base/Dockerfile.windows-2004, base/Dockerfile.windows-20H2
```

This fits what I want Renovate to do. Here's the renovate.json for reference

```json
{
  "extends": [
    "config:base",
    ":automergeDigest"
  ],
  "enabledManagers": ["dockerfile"],
  "dockerfile": {
    "ignorePaths": [
      "base/Dockerfile.aws",
      "scripts/",
      "scm/",
      "tools/",
      "msbuild/",
      "msbuild-2017/",
      "buildbot/",
      "ews/",
      "ews-legacy/"
    ],
    "pinDigests": true
  },
  "labels": [
    "renovate"
  ]
}
```

**Reply by rarkins** [reply]
Something else you could try is this:

```
{
  "extends": [
    "config:base",
    ":automergeDigest"
  ],
  "enabledManagers": ["dockerfile"],
  "dockerfile": {
    "includePaths": [
       "base/"
    ],
    "pinDigests": true
  },
  "packageRules": [{
    "paths": ["base/Dockerfile.aws"],
    "enabled": false
   }],
  "labels": [
    "renovate"
  ]
}
```

With the above it will parse everything in base/ but then disable all deps found in base/Dockerfile.aws

**Reply by donny-dont** [reply]
The `base/Dockerfile.1809` image and `base/Dockerfile.aws` are both `FROM mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2019`. Would that cause the `base/Dockerfile.1809` to not update?

**Reply by rarkins** [reply]
No, the rule I gave is based on file path and not image/dependency name.
